accepted/2023/dotnet-nuget-setapikey.md

```diff
@@ -0,0 +1,230 @@
+# dotnet nuget setapikey
+
+- Andy Zivkovic ([zivkan](https://github.com/zivkan))
+- Start Date (2022-12-15)
+- GitHub Issue (https://github.com/NuGet/Home/issues/6437)
+- GitHub PR (not yet implemented)
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+
+Add the `dotnet nuget apikey set` command, to allow customers to set a nuget.org api key in their `nuget.config` file. Also add `dotnet nuget apikey unset`, so customers can clean up secrets easily. Additionally, `nuget.config` will be changed to allow unencrypted (plain text) API keys.
+
+## Motivation
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+
+- `dotnet nuget apikey set`
+
+nuget.org is one of the only package sources that uses API keys (`X-NuGet-ApiKey` HTTP header), rather than source credentials (`Authorization` HTTP header).
```

Comment:
Hey @loic-sharma, it's been a long time!  Good to hear from you.

I went through the [list of NuGet server implementations in our docs for hosting private feeds](https://learn.microsoft.com/en-us/nuget/hosting-packages/overview), and I found about half of them use API key, and the others use package source credentials. (GitHub Package Repository surprised me, because [now they use API key](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-nuget-registry#publishing-a-package), but my memory is correct in that [they used to use package source credentials](http://web.archive.org/web/20210506213928/https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-nuget-registry)).

Anyway, I updated the paragraph, so it's more accurate now, and hopefully more clear as well.

Reply:
Gotcha, make sense. Thanks!